### ISSUE TITLE
P2P - Implement/Add Sync  Filter 

### ISSUE THREAD

**Issue by f-odhiambo** [opensrp/fhircore#1018]
- [ ]  Add Filter by Location, Practitioner, CarerTeams/ Organization

**Comment by Rkareko** [comment]
This is useful for down-syncing e.g CHSS -> CHA.

@f-odhiambo Is this implemented for the current server sync so that we can re-use the startegy/logic

**Comment by pld** [comment]
Wouldn't we want this to be generic? Not specific to P2P but used in both P2P and standard sync?

**Comment by f-odhiambo** [comment]
> This is useful for down-syncing e.g CHSS -> CHA.
> 
> @f-odhiambo Is this implemented for the current server sync so that we can re-use the strategy

Only sync by Org is available at the moment

**Comment by f-odhiambo** [comment]
> Wouldn't we want this to be generic? Not specific to P2P but used in both P2P and standard sync?

Good point I think we can review the sync use case ,I think current current sync works as follows, 1st time sync load everything from server based on org, subsequent sync fetch only updates
